src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/StreamPipeReader.cs

```diff
@@ -75,6 +75,11 @@ private CancellationTokenSource InternalTokenSource
         {
             get
             {
+                if (_internalTokenSource != null)
+                {
+                    return _internalTokenSource;
```

Comment:
This is broken if ClearCancellationToken is called concurrently.

Reply:
Oh, absolutely, good point. I completely missed the second usage!